include/seqan3/range/container/concatenated_sequences.hpp

```diff
@@ -181,7 +181,35 @@ class concatenated_sequences
     using size_type = ranges::size_type_t<data_delimiters_type>;
     //!\}
 
-    /* rule of six */
+protected:
+    //!\cond
+    // unfortunately we cannot specialise the variable template so we have to add an auxiliary here
+    template <typename t>
+        requires (dimension_v<t> == dimension_v<value_type> + 1) &&
+                 std::is_same_v<remove_cvref_t<innermost_value_type_t<value_type>>,
+                                remove_cvref_t<innermost_value_type_t<t>>>
+    static constexpr bool is_compatible_this_aux = true;
+    //!\endcond
+
+    //!\brief Whether a type satisfies seqan3::compatible_concept with this class.
+    //!\hideinitializer
+    // cannot use the concept, because this class is not yet fully defined
+    template <typename t>
+    static constexpr bool is_compatible_this = is_compatible_this_aux<t>                                    ||
```

Comment:
I don't fully understand this. Ahh `cannot use the concept, because this class is not yet fully defined`

Can't you encapsulate the `value type` inside some other class like `std::vector` (or an extra stub version) and check it with `compatible_concept`?!